source/PlayerInfo.cpp

```diff
@@ -1356,7 +1366,21 @@ void PlayerInfo::SetShipOrder(const vector<shared_ptr<Ship>> &newOrder)
 	// Check if the incoming vector contains the same elements
 	if(std::is_permutation(ships.begin(), ships.end(), newOrder.begin()))
 	{
+		auto oldFirstShip = ships[0];
```

Comment:
```suggestion
		std::shared_ptr<Ship> &oldFirstShip = ships.front();
```

Reply:
I have tried to implement it and I realised that it is not working. As this is now a reference to ships.front() but right after the vector changes and the reference is then referencing a wrong element.

Reply:
ah my fault, just remove the &